h/streamer/messages.py

```diff
@@ -140,8 +147,8 @@ def _generate_annotation_event(message, socket, annotation, user_nipsad):
     base_url = socket.registry.settings.get('h.app_url',
                                             'http://localhost:5000')
     links_service = LinksService(base_url, socket.registry)
-    serialized = presenters.AnnotationJSONPresenter(annotation,
-                                                    links_service).asdict()
+    resource = AnnotationResource(annotation, group_service)
```

Comment:
I think you could create the `GroupfinderService` here (`settings` is available as `socket.registry.settings`), which would obviate the need to pass settings into all the functions above.